x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -540,6 +542,46 @@ private static void updateExecutionInfoAfterCoordinatorOnlyQuery(EsqlExecutionIn
         }
     }
 
+    /**
+     * If all of target shards excluding the skipped shards failed, then we should fail the entire query regardless of the partial_results
+     * configuration or skip_unavailable setting. This behavior doesn't fully align with the search API as the skip_unavailable
+     * would ignore all the failures from the remote clusters; hence, only fail the request when all shards in the local cluster failed.
```

Comment:
It looks like it's looking for all clusters for failures, not just the local. Is the comment wrong, or the code, or I am misunderstanding what is supposed to happen? 

I tried it out, and this:
```
from remote1:existent | WHERE MATCH(id, "clothing", {"analyzer": "DOESNT_EXIST"})
```
produces an error just like the local version, even though local cluster is not the part of it.

Reply:
Yes, unlike the search API, we fail the query if all shards fail. I think I need to reword the comment.

Reply:
I've reworded this in https://github.com/elastic/elasticsearch/pull/131177/commits/a1f002ba84122defe763fbf2d6c96abd5adbc489. Let me know if it's clear.